### ISSUE TITLE
Cryptic "Bad Annotation" Error Message

### ISSUE THREAD

**Issue by ChadKillingsworth** [google/closure-compiler#2489]
A variable assignment with 2 `@extends` annotations gave this warning using NTI:

`WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.
`
That could be a lot better (more helpful).

```js
/**
  * @constructor
  * @extends {Polymer.Element}
  * @extends {ObservableModelsElement}
  */
const CardElement = observableModelsMixin(cardUtilsMixin(Polymer.GestureEventListeners(Polymer.Element)));
```

**Comment by dimvar** [comment]
I believe you would get the exact same error in the old type checker. I agree it would be nice to have a better explanation here.
